packages/dd-trace/src/profiling/profile.js

```diff
@@ -0,0 +1,132 @@
+'use strict'
+
+const { perftools } = require('../../../../proto/profile')
+
+class Profile extends perftools.profiles.Profile {
+  constructor (sampleType, periodType, period) {
+    super()
+
+    this._stringTable = new Map([['', 0]])
+    this._locations = new Map()
+    this._functions = new Map()
+    this._samples = new Map()
+    this._links = new Map()
+    this._sampleType = this._toSampleType(sampleType)
+    this._periodType = this._toValueType(periodType)
+    this._period = period
+    this._timeNanos = Date.now() * 1e6
+    this._properties = {}
+  }
+
+  export () {
```

Comment:
Because the internal structure is different. Also, it doesn't need to extend from perftools anymore, that's from a previous version of this class. I guess an alternative would be to keep the inheritance and add a method that updates the real fields with the intermediary fields.

Reply:
Actually I did that because the `perftools.profiles.Profile` class alters its constructor arguments. I'll remove the inheritance instead.